src/vs/workbench/common/editor.ts

```diff
@@ -1340,11 +1340,12 @@ export async function pathsToEditors(paths: IPathData[] | undefined, fileService
 			return;
 		}
 
-		const options: ITextEditorOptions = {
+		const options = {
+			...path.options,
 			selection: exists ? path.selection : undefined,
 			pinned: true,
 			override: path.editorOverrideId
-		};
+		} as IEditorOptions;
```

Comment:
I prefer to declare the type above in line 1343 over explicit cast:

```ts
const options: IEditorOptions = ...
```

Reply:
Yep, did the casing as object literal doesn't like unknown properties. Tweaked it.